source/Core/IOHandler.cpp

```diff
@@ -28,7 +28,9 @@
 #include "lldb/Core/State.h"
 #include "lldb/Core/StreamFile.h"
 #include "lldb/Core/ValueObjectRegister.h"
+#ifndef LLDB_DISABLE_LIBEDIT
```

Comment:
I basically fixed the build errors that I presume were caused by a bad merge - many syntax errors occured as a result - e.g. missing `{`, unhandled identifiers etc.

Take a look at this commit here: https://github.com/hughbe/swift-lldb/commit/8d544a47ba95533c369949fa3bf0ad01dadea0a3
It demonstrates what the diff looks like when I copy-and-paste the remaining bits of `IOHandler.cpp` from upstream. Basically, its changes in API that I didn't want to modify right now because I don't really wan't to break all the things!